src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,218 @@
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update "
+                    + car + " in cars table.", e);
+        }
+        deleteDriversFromCar(car.getId());
+        insertDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete driver with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id AS car_id, c.model AS car_model, "
+                + "c.manufacturer_id AS manufacturer_id, "
+                + "m.name AS manufacturer_name, m.country AS manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "INNER JOIN cars_drivers cd ON c.id = cd.car_id "
+                + "WHERE driver_id = ? AND c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't get a list of cars by driver id: " + driverId, e);
+        }
+        for (Car car: cars) {
+            car.setDrivers(getAllDriversByCarId(car.getId()));
+        }
+        return cars;
+    }
+
+    private void insertDriversToCar(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't write to cars_drivers table "
+                    + car + ". ", e);
+        }
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        long id = resultSet.getObject("car_id", Long.class);
```

Comment:
```suggestion
        Long id = resultSet.getObject("car_id", Long.class);
```
Use wrappers for ID field.